### PR TITLE
Adds info about DataListItem attribute

### PR DESCRIPTION
**Pull request by skttl**
### Description

I needed a way to extend the information about enums in data lists, and found it in the source code. So why not add some documentation? :)

### Types of changes

- [x] Documentation change
- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_

### Checklist

- [ ] My code follows the coding style of this project.
- [ ] My changes generate no new warnings.
- [ ] My change requires a change to the documentation.
- [x] I have updated the corresponding documentation.
- [x] I have read the **[CONTRIBUTING](CONTRIBUTING)** and **[CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)** documents.

### CODE REVIEW

File: docs/data-sources/data-source--enum.md [hunk 1]
```diff
@@ -22,3 +22,41 @@ Select the desired enumeration, by selecting the containing .NET assembly, and t
 The value returned from the List editor is the configured enumeration type.
 
 Depending on the `List editor` used, this may be wrapped in a `List<T>`.
+
+##### Controlling Enum display in Contentment
+
+Here's the proofread version of the text:
+
```
Comment (leekelleher) [inline docs/data-sources/data-source--enum.md:29]: We can remove the "proofread" bit.

File: docs/data-sources/data-source--enum.md [hunk 2]
```diff
@@ -22,3 +22,41 @@ Select the desired enumeration, by selecting the containing .NET assembly, and t
 The value returned from the List editor is the configured enumeration type.
 
 Depending on the `List editor` used, this may be wrapped in a `List<T>`.
+
+##### Controlling Enum display in Contentment
```
Comment (leekelleher) [inline docs/data-sources/data-source--enum.md:26]: I'd remove the "in Contentment" bit, as the documentation is already in context of the package.

File: docs/data-sources/data-source--enum.md [hunk 3]
```diff
@@ -22,3 +22,41 @@ Select the desired enumeration, by selecting the containing .NET assembly, and t
...
+|-------------|------------------------------------------------------------------|---------|
+| Description | Provides a description for the Enum value                         | string  |
+| Disabled    | Specifies whether the Enum value should be disabled               | bool    |
+| Group       | Assigns the Enum value to a specific group                        | string  |
+| Icon        | Specifies an icon for the Enum value                              | string  |
+| Ignore      | Indicates whether the Enum value should be ignored                | bool    |
+| Name        | Provides a custom name for the Enum value (default: field name)   | string  |
+| Value       | Specifies a custom value for the Enum value (default: field value)| string  |
+
+The provided text has been proofread and revised for clarity and accuracy.
```
Comment (leekelleher) [inline docs/data-sources/data-source--enum.md:62]: We can remove this "proofread" bit too.

**Review by leekelleher — CHANGES_REQUESTED** [review]
@skttl Thank you very much for the documentation update, great addition.
Apart from a couple of small removals, all good!

**Comment by skttl** [comment]
Thanks, I've updated :)

**Review by leekelleher — APPROVED** [review]
Super, thank you @skttl!

**Pull request merged**